frontend/packages/knative-plugin/src/utils/create-eventsources-utils.ts

```diff
@@ -18,8 +20,38 @@ import * as camelSourceImg from '../imgs/logos/camelsource.svg';
 import * as containerSourceImg from '../imgs/logos/containersource.png';
 import * as cronJobSourceImg from '../imgs/logos/cronjobsource.png';
 import * as kafkaSourceImg from '../imgs/logos/kafkasource.svg';
+import CronJobSection from '../components/add/event-sources/CronJobSection';
+import SinkBindingSection from '../components/add/event-sources/SinkBindingSection';
+import ApiServerSection from '../components/add/event-sources/ApiServerSection';
+import KafkaSourceSection from '../components/add/event-sources/KafkaSourceSection';
+// eslint-disable-next-line import/no-cycle
+import YAMLEditorSection from '../components/add/event-sources/YAMLEditorSection';
 
-export const getEventSourcesDepResource = (formData: EventSourceFormData): K8sResourceKind => {
+export const isKnownEventSourceSelected = (eventSource: string): boolean =>
+  !!Object.values(KnownEventSources).find((knownEventSource) => eventSource === knownEventSource);
+
+const getResourceFromYAML = (formData: any): K8sResourceKind => {
+  const {
+    application: { name: applicationName },
+    yamlData,
+  } = formData;
+  const eventYaml = safeLoad(yamlData);
+  const { metadata } = eventYaml;
+  const defaultLabel = getAppLabels(metadata.name, applicationName);
+  const eventSourceResource: K8sResourceKind = {
+    ...eventYaml,
+    metadata: {
+      ...metadata,
+      labels: {
+        ...defaultLabel,
+      },
+      annotations,
+    },
+  };
+  return eventSourceResource;
+};
```

Comment:
why we need this , we have `yamlData` in initial values, can't that be used?

Reply:
fixed